crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs

```diff
@@ -2268,6 +2714,270 @@ fn test() {
         // ),
     ];
 
+    let fail_typescript = vec![
+        ("import foo from 'import1';", Some(serde_json::json!(["import1", "import2"]))),
+        ("import foo = require('import1');", Some(serde_json::json!(["import1", "import2"]))),
+        ("export { foo } from 'import1';", Some(serde_json::json!(["import1", "import2"]))),
+        (
+            "import foo from 'import1';",
+            Some(serde_json::json!([{ "paths": ["import1", "import2"] }])),
+        ),
+        (
+            "export { foo } from 'import1';",
+            Some(serde_json::json!([{ "paths": ["import1", "import2"] }])),
+        ),
+        // (
```

Comment:
`paths` and `patterns` combination is currently not supported. Looks like a typescript-eslint only feature.